src/main/java/org/alfresco/mockeventgenerator/CamelMessageProducer.java

```diff
@@ -82,21 +84,27 @@ public void send(Object message, Map<String, Object> headers) throws Exception
             {
                 aggregated.set(true);
             }
-            sendImpl(message, headers);
+            sendImpl(message, headers, endpoint);
         }
     }
 
-    private void sendImpl(Object message, Map<String, Object> headers) throws Exception
+    private String getEndpointByKey(String key)
```

Comment:
Should this throw an error if the key is not found?

And what if the key is null or empty?